conngen/conngen.cpp

```diff
@@ -37,185 +34,312 @@
 // Includes from sli:
 #include "token.h"
 
+namespace nest
+{
+
 void
-nest::cg_connect( nest::ConnectionGeneratorDatum& cg,
-  const index source_id,
-  const index target_id,
+cg_connect( ConnectionGeneratorDatum& cg,
+  const GIDCollection& source_gids,
+  const GIDCollection& target_gids,
   const DictionaryDatum& params_map,
   const Name& synmodel_name )
 {
-  Subnet* sources =
-    dynamic_cast< Subnet* >( kernel().node_manager.get_node( source_id ) );
-  if ( sources == NULL )
-  {
-    LOG( M_ERROR, "CGConnect_cg_i_i_D_l", "sources must be a subnet." );
-    throw SubnetExpected();
-  }
-  if ( not sources->is_homogeneous() )
-  {
-    LOG( M_ERROR,
-      "CGConnect_cg_i_i_D_l",
-      "sources must be a homogeneous subnet." );
-    throw BadProperty();
-  }
-  if ( dynamic_cast< Subnet* >( *sources->local_begin() ) )
+  const Token synmodel =
+    kernel().model_manager.get_synapsedict()->lookup( synmodel_name );
+  if ( synmodel.empty() )
   {
-    LOG( M_ERROR,
-      "CGConnect_cg_i_i_D_l",
-      "Only 1-dim subnets are supported as sources." );
-    throw BadProperty();
+    throw UnknownSynapseType( synmodel_name.toString() );
   }
+  const index synmodel_id = static_cast< index >( synmodel );
+
+  RangeSet source_ranges, target_ranges;
+
+  cg_get_ranges( source_ranges, source_gids );
+  cg_get_ranges( target_ranges, target_gids );
+  cg_set_masks( cg, source_ranges, target_ranges );
```

Comment:
As far as I can see, `source_ranges` and `target_ranges` are only created to be passed to `cg_set_masks()`, but not used in subsequent code. Wouldn't it make code here more easily readable if the `cg_get_ranges()` calls were moved into `cg_set_masks()`, which would then just receive the `*_gids`?

Reply:
Fixed in fe4408921.